core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -1003,18 +1002,6 @@ public FeaturesConfig setIgnoreDownstreamPreferences(boolean ignoreDownstreamPre
         return this;
     }
 
-    public boolean isIterativeRuleBasedColumnPruning()
```

Comment:
Do we need to add this to list of defunct configs? I'm not sure what's the policy for other such optimizer related configs.

Reply:
> it seems that redundant projections are added

Some of the added projections are pruning, but indeed, some are identity. Thanks for pointing that out.

After investigating the new projections, it seems to me that they all result from enhanced capability of the rule set as compared to the legacy optimizer to prune the `WindowNode`'s source:
- whenever a pruning projection appears under `WindowNode`, it means that the pruning couldn't be pushed down any more
- whenever an identity projection appears under `WindowNode`, it means that a rule inserted a pruning projection, which was further absorbed by the upstream plan (typically, by a `TableScanNode`), making the projection effectively identity. Please note that neither the rule set, nor the legacy optimizer remove identity projections. Only in the legacy scenario the projections didn't appear at all.

To clear the test cases from the identity projections, I followed your suggestion, and made sure that the pruning rules run along with `RemoveRedundantIdentityProjections`. 

The remaining new projections (pruning) had to be added to the expected plans though. I added an explanatory comment in the commit message.